### ISSUE TITLE
Can't create more than 1 template

### ISSUE THREAD

**Issue by orangebutblue** [KenediaDev/Kenedia.BlishHUD#26]
The + button has absolutely no effect, when I press it.

If I copy a build to clipboard and press the + button: again, nothing happens.

If I right-click the single template that I have and press "duplicate" then all of blish completely crashes

here are the blish logs:

```
09:50:12.2286 +01:00 | FATAL | Blish_HUD.DebugService | Blish HUD encountered a fatal crash!
System.NullReferenceException: Object reference not set to an instance of an object.
   at Kenedia.Modules.BuildsManager.Controls.Selection.TemplateSelectable.ApplyTemplate() in C:\BhApps\BhudRequestFor\work\Kenedia.Modules.BuildsManager-4l30XP72\BuildsManager\Controls\Selection\TemplateSelectable.cs:line 538
   at Kenedia.Modules.BuildsManager.Controls.Selection.TemplateSelectable..ctor(TemplatePresenter templatePresenter, TemplateCollection templates, Data data, TemplateTags templateTags, TemplateFactory templateFactory) in C:\BhApps\BhudRequestFor\work\Kenedia.Modules.BuildsManager-4l30XP72\BuildsManager\Controls\Selection\TemplateSelectable.cs:line 161
   at Kenedia.Modules.BuildsManager.Controls.Selection.BuildSelection.AddTemplateSelectable(Boolean firstLoad, List`1 addedTemplates) in C:\BhApps\BhudRequestFor\work\Kenedia.Modules.BuildsManager-4l30XP72\BuildsManager\Controls\Selection\BuildSelection.cs:line 310
   at Kenedia.Modules.BuildsManager.Controls.Selection.BuildSelection.Templates_CollectionChanged(Object sender, NotifyCollectionChangedEventArgs e) in C:\BhApps\BhudRequestFor\work\Kenedia.Modules.BuildsManager-4l30XP72\BuildsManager\Controls\Selection\BuildSelection.cs:line 283
   at System.Collections.Specialized.NotifyCollectionChangedEventHandler.Invoke(Object sender, NotifyCollectionChangedEventArgs e)
   at System.Collections.ObjectModel.ObservableCollection`1.OnCollectionChanged(NotifyCollectionChangedEventArgs e)
   at System.Collections.ObjectModel.ObservableCollection`1.InsertItem(Int32 index, T item)
   at Kenedia.Modules.BuildsManager.Models.TemplateCollection.Add(Template template) in C:\BhApps\BhudRequestFor\work\Kenedia.Modules.BuildsManager-4l30XP72\BuildsManager\Models\TemplateCollection.cs:line 53
   at Kenedia.Modules.BuildsManager.Controls.Selection.TemplateSelectable.DuplicateTemplate() in C:\BhApps\BhudRequestFor\work\Kenedia.Modules.BuildsManager-4l30XP72\BuildsManager\Controls\Selection\TemplateSelectable.cs:line 0
   at Blish_HUD.Controls.Control.OnLeftMouseButtonReleased(MouseEventArgs e) in D:\a\Blish-HUD\Blish-HUD\Blish HUD\Controls\Control.cs:line 214
   at Blish_HUD.Controls.Control.TriggerMouseInput(MouseEventType mouseEventType, MouseState ms) in D:\a\Blish-HUD\Blish-HUD\Blish HUD\Controls\Control.cs:line 849
   at Blish_HUD.Controls.Container.TriggerMouseInput(MouseEventType mouseEventType, MouseState ms) in D:\a\Blish-HUD\Blish-HUD\Blish HUD\Controls\Container.cs:line 226
   at Blish_HUD.Controls.Container.TriggerMouseInput(MouseEventType mouseEventType, MouseState ms) in D:\a\Blish-HUD\Blish-HUD\Blish HUD\Controls\Container.cs:line 226
   at Blish_HUD.Input.MouseHandler.Update() in D:\a\Blish-HUD\Blish-HUD\Blish HUD\GameServices\Input\Mouse\MouseHandler.cs:line 164
   at Blish_HUD.InputService.Update(GameTime gameTime) in D:\a\Blish-HUD\Blish-HUD\Blish HUD\GameServices\InputService.cs:line 68
   at Blish_HUD.BlishHud.Update(GameTime gameTime) in D:\a\Blish-HUD\Blish-HUD\Blish HUD\BlishHud.cs:line 136
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime)
   at Microsoft.Xna.Framework.Game.Tick()
   at MonoGame.Framework.WinFormsGameWindow.TickOnIdle(Object sender, EventArgs e)
   at System.Windows.Forms.Application.ThreadContext.System.Windows.Forms.UnsafeNativeMethods.IMsoComponent.FDoIdle(Int32 grfidlef)
   at System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
   at MonoGame.Framework.WinFormsGameWindow.RunLoop()
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior)
   at Blish_HUD.Program.Main(String[] args) in D:\a\Blish-HUD\Blish-HUD\Blish HUD\Program.cs:line 92

``` 

I don't really know what `D:\a\Blish-HUD\Blish-HUD\Blish HUD\` is. I don't have that folder on my d drive.

**Comment by orangebutblue** [comment]
By "newest version" you mean 1.0.5, right? I have that.

To make it easier for you, I've disabled all of my other blish modules and then restarted blish to get a new log file. Then I activated builds manager.

When I press the + button (with or without a build copied in my clipboard) nothing happens - it doesn't even produce a log error. But when I right click any of my two builds (not sure why I have 2 now, it used to be only 1), then blish completely crashes.

Basically I've turned on the module, clicked + a couple of times, then did a right-click -> duplicate. That's the log for that:
[blishhud.20241030-201231.log](https://github.com/user-attachments/files/17577841/blishhud.20241030-201231.log)

**Comment by ejbaker** [comment]
Same issue. I can't figure out how to "save" the template I currently have active, which means I also can't make a new template. I tried deleting the default template, only to find that the "+" button doesn't do anything. Duplicating the first template causes Blish to hang for a moment, then crash.

**Comment by KenediaDev** [comment]
@orangebutblue @ejbaker Could any of you contact me on discord / share the content of your `\Guild Wars 2\addons\blishhud\kenedia\builds_manager\templates` folder? I can create, duplicate and delete templates without an issue. 

If one of you could hit me up on discord so we can figure it out together, that would be great!

**Comment by KenediaDev** [comment]
I've released a new version. Let me know if **v.1.0.6** fixed the issue.

**Comment by orangebutblue** [comment]
The new version is better, but there are a lot of other issues now 😄 

It's an extremely weird behavior. I could make a video or share my screen with you on discord.
whats your discord name so I can contact you? Or is there an official discord server?

**Comment by KenediaDev** [comment]
I would love to see the issues!

https://discord.gg/FYKN3qh
you can find me there as Kenedia (same avatar)
